src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Windows.cs

```diff
@@ -1246,5 +1284,19 @@ internal static SocketError Disconnect(Socket socket, SafeSocketHandle handle, b
 
             return errorCode;
         }
+
+        internal static unsafe SocketError DuplicateSocket(SafeSocketHandle handle, int targetProcessId, out SocketInformation socketInformation)
+        {
+            socketInformation = new SocketInformation
+            {
+                ProtocolInformation = new byte[Interop.Winsock.WSAProtocolInfo.Size]
```

Comment:
nit: stackalloc instead of newing up this array

Reply:
This is an [existing public API](https://docs.microsoft.com/en-us/dotnet/api/system.net.sockets.socketinformation.protocolinformation), that uses `byte[]`.